src/AppInstallerCLICore/Workflows/PortableInstallFlow.cpp

```diff
@@ -0,0 +1,309 @@
+                    return Runtime::GetPathTo(Runtime::PathName::PortableLinksMachineLocationX64);
+                }
+            }
+            else
+            {
+                return Runtime::GetPathTo(Runtime::PathName::PortableLinksUserLocation);
+            }
+        }
+
+        Manifest::AppsAndFeaturesEntry GetAppsAndFeaturesEntryForPortableInstall(std::vector<AppInstaller::Manifest::AppsAndFeaturesEntry> appsAndFeaturesEntries, AppInstaller::Manifest::Manifest& manifest)
+        {
+            AppInstaller::Manifest::AppsAndFeaturesEntry appsAndFeaturesEntry;
+            std::string packageId = manifest.Id;
+            std::string displayVersion = manifest.Version;
+            std::string displayName = manifest.DefaultLocalization.Get<Manifest::Localization::PackageName>();
+            std::string publisher = manifest.DefaultLocalization.Get<Manifest::Localization::Publisher>();
+
+            if (appsAndFeaturesEntries.empty())
+            {
+                appsAndFeaturesEntry = {};
+            }
+            else
+            {
+                appsAndFeaturesEntry = appsAndFeaturesEntries[0];
+            }
+
+            if (appsAndFeaturesEntry.DisplayName.empty())
+            {
+                appsAndFeaturesEntry.DisplayName = displayName;
+            }
+            if (appsAndFeaturesEntry.DisplayVersion.empty())
+            {
+                appsAndFeaturesEntry.DisplayVersion = displayVersion;
+            }
+            if (appsAndFeaturesEntry.Publisher.empty())
+            {
+                appsAndFeaturesEntry.Publisher = publisher;
+            }
+            if (appsAndFeaturesEntry.ProductCode.empty())
+            {
+                appsAndFeaturesEntry.ProductCode = packageId;
+            }
+
+            return appsAndFeaturesEntry;
+        }
+
+        void AppendExeExtension(std::string& value)
+        {
+            if (!HasSuffix(value, ".exe"))
+            {
+                value += ".exe";
+            }
+        }
+
+        bool AddPortableLinksDirToPathRegistry(Manifest::ScopeEnum& scope, Utility::Architecture& arch, bool& isPathModified)
+        {
+            Key key;
+            if (scope == Manifest::ScopeEnum::Machine)
+            {
+                key = Registry::Key::CreateKeyAndOpen(HKEY_LOCAL_MACHINE, Normalize(s_PathSubkey_Machine));
+            }
+            else
+            {
+                key = Registry::Key::CreateKeyAndOpen(HKEY_CURRENT_USER, Normalize(s_PathSubkey_User));
+            }
+
+            std::string portableLinksDir = GetPortableLinksLocation(scope, arch).u8string();
+            AICLI_LOG(CLI, Info, << "Adding to Path environment variable: " << portableLinksDir);
+            std::wstring pathKey = { L"Path" };
+            std::string pathValue = key[pathKey]->GetValue<Value::Type::String>();
+            if (pathValue.find(portableLinksDir) == std::string::npos)
+            {
+                std::string modifiedPathValue = pathValue;

```

Comment:
I don't see a reason to copy the string to modify it; why do you need the previous path string?